vendor/bat-native-ads/src/bat/ads/internal/p2a/p2a.cc

```diff
@@ -0,0 +1,101 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/p2a/p2a.h"
+
+#include <stdint.h>
+#include <string>
+
+#include "base/json/json_writer.h"
+#include "bat/ads/internal/ads_impl.h"
+#include "bat/ads/internal/classification/classification_util.h"
+
+namespace ads {
+
+P2A::P2A(
+    AdsImpl* ads)
+    : ads_(ads) {
+  DCHECK(ads_);
+}
+
+P2A::~P2A() = default;
+
+void P2A::RecordEventForAdOpportunity(
+    const std::vector<std::string>& segments) {
+  std::vector<std::string> questions_list;
+  std::vector<std::string> parent_segments =
+      classification::GetParentCategories(segments);
+
+  for (auto& segment : parent_segments) {
+    // Assume all segments are lower case
+    segment.erase(remove(segment.begin(), segment.end(), ' '), segment.end());
+    segment.erase(remove(segment.begin(), segment.end(), '&'), segment.end());
```

Comment:
You can use `segment.erase(std::remove_if(d.begin(), d.end(), [](char c) { return !std::isalnum(c); }), d.end());` to remove all non alpha